packages/test-snaps/src/features/snaps/get-entropy/hooks/useEntropySelector.tsx

```diff
@@ -0,0 +1,59 @@
+import type { EntropySource } from '@metamask/snaps-sdk';
+import { useState } from 'react';
+
+import { Tag, useInvokeQuery } from '../../../../api';
+import { getSnapId, useInstalled } from '../../../../utils';
+import { EntropySelector } from '../components';
+
+export type UseEntropySelectorOptions = {
+  /**
+   * Whether to show the raw list of entropy sources.
+   */
+  raw?: boolean;
+
+  /**
+   * The snap ID to use for the entropy sources.
+   */
+  snapId: `npm:${string}`;
+
+  /**
+   * The port to use for the entropy sources.
+   */
+  port: number;
+};
+
+/**
+ * Use a selector to select the entropy source to use.
+ *
+ * @param options - The options to use.
+ * @param options.snapId - The snap ID to use for the entropy sources.
+ * @param options.port - The port to use for the entropy sources.
+ * @param options.raw - Whether to show the raw list of entropy sources.
+ * @returns The entropy source and selector.
+ */
+export const useEntropySelector = ({
+  snapId: publicSnapId,
+  port,
+  raw = false,
+}: UseEntropySelectorOptions) => {
+  const [source, setSource] = useState<string | undefined>(undefined);
+  const snapId = getSnapId(publicSnapId, port);
```

Comment:
Not sure I fully follow the "translation" of the ID here?

Reply:
`getSnapId` is the helper function we use everywhere, which returns the NPM Snap ID (`publicSnapId` in this case) if running on the hosted version, or the local Snap ID if running locally.